binding/dotnet/PorcupineTest/MainTest.cs

```diff
@@ -63,20 +114,50 @@ public void TestProcess()
                 int start = i * frameLen;
                 int count = frameLen;
                 List<short> frame = data.GetRange(start, count);
-                int result = p.Process(frame.ToArray());
-                Assert.IsTrue(result == -1 || result == 0, "Porcupine returned an unexpected result (should return 0 or -1)");
+                int result = porcupine.Process(frame.ToArray());
+                Assert.IsTrue(result >= -1, "Porcupine returned an unexpected result (<-1)");
                 if (result >= 0)
                 {
                     results.Add(result);
                 }
             }
 
-            Assert.IsTrue(results.Count == 1 && results[0] == 0, "Expected to find keyword 'porcupine', but no keyword was detected.");
-            p.Dispose();
+            Assert.AreEqual(expectedResults.Count, results.Count, $"Should have found {expectedResults.Count} keywords, but {results.Count} were found.");
+            for (int i = 0; i < results.Count; i++)
+            {
+                Assert.AreEqual(expectedResults[i], results[i], $"Porcupine did not detect keyword #{i+1} correctly.");
```

Comment:
Always best to print any relevant info to help with debugging:
```suggestion
                Assert.AreEqual(expectedResults[i], results[i], $"Expected keyword {expectedResults[i]}, but Porcupine detected keyword {results[i]}.");
```